WooCommerce/WooCommerceTests/POS/Analytics/POSCollectOrderPaymentAnalyticsTests.swift

```diff
@@ -0,0 +1,36 @@
+@testable import WooCommerce
+import protocol WooFoundation.Analytics
+import Testing
+
+struct POSCollectOrderPaymentAnalyticsTests {
+    private let analytics: Analytics
+    private let analyticsProvider: MockAnalyticsProvider
+
+    init() {
+        analyticsProvider = MockAnalyticsProvider()
+        analytics = WooAnalytics(analyticsProvider: analyticsProvider)
+    }
+
+    @Test func POSCollectOrderPaymentAnalyticsTests_when_successful_payment_then_tracks_event_and_properties() {
+        // Given
```

Comment:
I think this test is fine 👍 

To have tests that calculate event timing logic, we would need for `POSCollectOrderPaymentAnalytics` methods or initializer to inject some date handler that would allow us to inject specific date instead of `Date()`.